libraries/AP_NavEKF3/AP_NavEKF3_Outputs.cpp

```diff
@@ -306,7 +306,7 @@ bool NavEKF3_core::getLLH(struct Location &loc) const
 {
     if(validOrigin) {
         // Altitude returned is an absolute altitude relative to the WGS-84 spherioid
-        loc.alt = EKF_origin.alt - outputDataNew.position.z*100;
+        loc.alt =  100 * (int32_t)(ekfGpsRefHgt - (double)outputDataNew.position.z);
```

Comment:
Thinking about it some more, we should also adjust the altitude reported by getPosD for the difference between EKF_Origin.alt and ekfGpsRefHgt if internal reference height adjustments have been requested by the setting of bit positions 0 and 1.